pkg/kubelet/rkt/config.go

```diff
@@ -33,24 +39,46 @@ type Config struct {
 	// Comma-separated list of security features to disable.
 	// Allowed values: "none", "image", "tls", "ondisk", "http", "all".
 	InsecureOptions string
-	// The local config directory.
-	LocalConfigDir string
+	// Local configuration directory.
+	LocalConfig string
+	// System configuration directory.
+	SystemConfig string
+	// Automatically trust gpg keys fetched from https.
+	TrustKeysFromHttps bool
+	// User configuration directory.
+	UserConfig string
+
+	// The Option strings. This string passed to rkt directly.
+	Options string
 }
 
-// buildGlobalOptions returns an array of global command line options.
-func (c *Config) buildGlobalOptions() []string {
-	var result []string
-	if c == nil {
-		return result
+func NewConfig(path, stage1Image, options string) (*Config, error) {
+	fs := flag.NewFlagSet("rktflags", flag.ContinueOnError)
+	config := &Config{
+		Path:        path,
+		Stage1Image: stage1Image,
+		Options:     options,
 	}
 
-	result = append(result, fmt.Sprintf("--debug=%v", c.Debug))
-	result = append(result, fmt.Sprintf("--insecure-options=%s", c.InsecureOptions))
-	if c.LocalConfigDir != "" {
-		result = append(result, fmt.Sprintf("--local-config=%s", c.LocalConfigDir))
+	fs.BoolVar(&config.Debug, "debug", false, "print out more debug information to stderr")
+	fs.StringVar(&config.Dir, "dir", "", "rkt data directory")
+	fs.StringVar(&config.InsecureOptions, "insecure-options", "", `comma-separated list of security features to disable. Allowed values: "none", "image", "tls", "ondisk", "http", "all"`)
+	fs.StringVar(&config.LocalConfig, "local-config", "", "local configuration directory")
+	fs.StringVar(&config.SystemConfig, "system-config", "", "system configuration directory")
+	fs.BoolVar(&config.TrustKeysFromHttps, "trust-keys-from-https", false, "automatically trust gpg keys fetched from https")
+	fs.StringVar(&config.UserConfig, "user-config", "", "user configuration directory")
+
+	if err := fs.Parse(strings.Fields(options)); err != nil {
+		return nil, err
 	}
-	if c.Dir != "" {
-		result = append(result, fmt.Sprintf("--dir=%s", c.Dir))
+	return config, nil
+}
+
+// buildGlobalOptions returns an array of global command line options.
+func (c *Config) buildGlobalOptions() []string {
+	result := strings.Fields(c.Options)
+	if c.Stage1Image != "" {
+		result = append(result, fmt.Sprintf("--stage1-path=%s", c.Stage1Image))
```

Comment:
Should we check rkt version here, if we still use deprecated `stage1Image `field? 
IMHO it'll fail, if we use rkt build below 1.0, because of lack` --stage1-path` flag (is there a still need support for rkt < 1.0 in k8).